creator/nx_creator_ptycho.py

```diff
@@ -117,44 +118,43 @@ def _create_dataset(self,
 
     def _check_unit(self, name, expected, supplied):
         """
-        Unit check for supplied units
+        Return `True` if conversion is possible between expected and supplied units.
 
-        Our test for units should check if the supplied
-        units string can be mapped into the expected units for that field.
         If arbitrary units are supplied in form of 'au', 'a.u.' or 'a.u' no conversion is applied
-        and pint if not used for the unit check.
-
+        and pint is not used for the unit check.
         :param : name of field
         :param : expected units
         :param : units string that was given
-        :return *bool*: `True` if units conversion is possible:
+        :return *bool*: `True` if units conversion is possible
         """
 
         # catch arbitrary unit separately from pint --> point that out in documentation
         if supplied in ['au', 'a.u.', 'a.u']:
-            logger.info("Info: arbitrary units supplied for '%s' in form of '%s' np unit conversion applicable",
+            logger.info("Arbitrary units supplied for '%s' in form of '%s' no unit conversion or "
+                        "pint unit check applicable",
                         name,
                         supplied)
             return True
         else:
             ureg = pint.UnitRegistry()
-            user = 1.0 * ureg(supplied)
             try:
-                user.to(expected)
+                user = 1.0 * ureg(supplied)
+            except pint.UndefinedUnitError as err:
+                print(f'WARNING: {err} --> units for {name} not written')
+                return
+            if user.check(expected):
+                print(f"Units {supplied} added to {name}")
                 return True
-            except pint.DimensionalityError:
-                logger.warning("WARNING: '%s': Supplied unit (%s) does not match expected units (%s)",
-                               name,
-                               supplied,
-                               expected)
+            else:
+                print(f'WARNING: Supplied unit [{supplied}] for {name} does not match expected units [{expected}]')
```

Comment:
This section is much better. It catches more cases. But why use print statements instead of the warnings module?

Reply:
Good catch.  Will *warnings* only post this notice once per process?  We want to see the warning every time this part of the code is executed.

Reply:
Yes. Warnings post once by default, but we can override that behavior. https://stackoverflow.com/a/22661838/4459405

Reply:
Seems a custom PostAlways warning is a solution.

Reply:
Ok, so I tried to use the `warnings.warn`. However `warnings.warn` can't handle f-strings (and variables in string messages in general just like `print` can do)
Thus I use `logger.warn` now, which might be the better choice for our intended use of printed feedback to the user in this case anyways.
Thoughts?